src/widgets/chart.rs

```diff
@@ -545,12 +551,17 @@ impl<'a> Widget for Chart<'a> {
             Block::default()
                 .borders(Borders::ALL)
                 .render(legend_area, buf);
-            for (i, dataset) in self.datasets.iter().enumerate() {
+            for (i, (dataset_name, dataset_style)) in self
+                .datasets
+                .iter()
+                .filter_map(|d| Some((d.name.as_ref()?, d.style)))
```

Comment:
I think this returns Some regardless of whether the item has a name and hence doesn't filter out empty names.